packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1617,8 +1618,19 @@ export class ContainerRuntime
 			return this.submitSignalFn(envelope2, targetClientId);
 		};
 
+		let snapshot: ISnapshot | ISnapshotTree | undefined = getSummaryForDatastores(
+			baseSnapshot,
+			metadata,
+		);
+		if (snapshot !== undefined && snapshotWithContents !== undefined) {
```

Comment:
minor: I think if we have snapshotWithContents , we should always have snapshot, not? At a minimum, we do not need it, as we can get everything we want from snapshotWithContents , right?

Reply:
Yes, but we are calculating actual snapshot using:
`etSummaryForDatastores(
			baseSnapshot,
			metadata,
		);`
That is why I have kept that outside if condition as that is common. Then in this If, I have to check both just for the compiler sake.